scrapy/downloadermiddlewares/httpcache.py

```diff
@@ -24,6 +29,8 @@ def __init__(self, settings, stats):
         self.ignore_missing = settings.getbool('HTTPCACHE_IGNORE_MISSING')
         self.stats = stats
 
+        logger.debug("Using cache directory %(cachedir)s" % {'cachedir': self.storage.cachedir})
```

Comment:
All built-in cache storages have 'cachedir' attribute, but it is not guaranteed for third-party cache storages. I think it could be better to either check that this attribute is present, or (better) move logging to storages. The latter will allow to log not only directory, but also a file name - for some storages cache is stored in a single file.